episodes/2-identify-the-problem.Rmd

```diff
@@ -42,62 +35,57 @@ rodents <- read_csv("data/surveys_complete_77_89.csv")
 ```
- 
- 
- Whoops! Here we see another error message -- this time, R responds with a perhaps more-uninterpretable message.
+
+Whoops! Here we see another error message -- this time, R responds with a perhaps more-uninterpretable message.
 
 Let's go over each part briefly. First, we see an error from a function called `fortify`, which we didn't even call! Then a much more helpful informational message: Did we accidentally pass `aes()` to the `data` argument? This does seem to relate to our function call, as we do pass `aes`, potentially where our data should go. A helpful starting place when attempting to decipher an error message is checking the documentation for the function which caused the error:
 
-` ?ggplot`
+`?ggplot`
 
 Here, a Help window pops up in RStudio which provides some more information. Skipping the general description at the top, we see ggplot takes positional arguments of `data`, **then** `mapping`, which uses the `aes` call. We can see in "Arguments" that the `aes(x = rodents$taxa)` object used in the plot is attempted by `fortify` to be converted to a data.frame: now the picture is clear! We accidentally passed our `mapping` argument (telling ggplot how to map variables to the plot) into the position it expected `data` in the form of a data frame. And if we scroll down to "Examples", to "Pattern 1", we can see exactly how ggplot expects these arguments in practice. Let's amend our result:
 
-```{r} 
+```{r}
 ggplot(rodents, aes(x = taxa)) + geom_bar()
 ```
 
+Here we see our desired plot.
 
-Here we see our desired plot. 
-
-
+Some common errors you might run into include R expecting different input types for a function than those given, being unable to process data in the way the code is written (e.g. divide by zero errors), or being unable to find a variable or function you've specified.
 
 ## Summary
 
 In general, when encountering an error message for which a remedy is not immediately apparent, some steps to take include:
 
-1. Reading each part of the error message, to see if we can interpret and act on any.
+1.  Reading each part of the error message, to see if we can interpret and act on any.
 
-2. Pulling up the R Documentation for that function (which may be specific to a package, such as with ggplot).
+2.  Pulling up the R Documentation for that function (which may be specific to a package, such as with ggplot).
 
-3. Reading through the documentation's Description, Usage, Arguments, Details and Examples entries for greater insight into our error. 
+3.  Reading through the documentation's Description, Usage, Arguments, Details and Examples entries for greater insight into our error.
 
-4. Copying and pasting the error message into a search engine for more interpretable explanations.
+4.  Copying and pasting the error message into a search engine for more interpretable explanations.
 
 And, when all else fails, preparing our code into a reproducible example for expert help.
```

Comment:
Right, so this speaks to two of my comments from before. These steps show the structure/flow of this lesson, proceeding from trying to decipher/fix the error message directly to creating a reprex if that proves impossible or unhelpful.

Changing the wording at the beginning to clarify that, and keeping this eventuality in mind no matter how we decide to handle jargon, will help to reinforce this narrative.

Reply:
I don't understand the latter half of this comment ("and keeping this eventuality in mind no matter how we decide to handle jargon, will help to reinforce this narrative") but I think it makes sense overall.